src/components/migrated-bch.js

```diff
@@ -0,0 +1,324 @@
+import React, { Component } from 'react';
+import { InputField } from './form-components';
+import { Form, Button, Row, Col, Alert } from 'reactstrap';
+import { address, HDNode, Transaction, TransactionBuilder } from 'bitgo-utxo-lib';
+
+import * as _ from 'lodash';
+
+import ErrorMessage from './error-message';
+
+import tooltips from 'constants/tooltips';
+
+import moment from 'moment';
+
+const fs = window.require('fs');
+const { dialog } = window.require('electron').remote;
+const formTooltips = tooltips.migratedBch;
+
+class MigratedBchRecoveryForm extends Component {
+  state = {
+    walletId: '',
+    recoveryAddress: '',
+    passphrase: '',
+    prv: '',
+    recoveryTx: null,
+    logging: [''],
+    error: '',
+    recovering: false,
+    twofa: '',
+  };
+
+  collectLog = (...args) => {
+    const { logging } = this.state;
+    const newLogging = logging.concat(args);
+    this.setState({ logging: newLogging });
+  };
+
+  updateRecoveryInfo = (field) => (value) => {
+    this.setState({ [field]: value });
+  };
+
+  resetRecovery = () => {
+    this.setState({
+      walletId: '',
+      tokenAddress: '',
+      recoveryAddress: '',
+      passphrase: '',
+      prv: '',
+      recoveryTx: null,
+      logging: [''],
+      error: '',
+      recovering: false,
+      twofa: '',
+    });
+  };
+
+  createRecoveryTx = async (bch, migratedWallet) => {
```

Comment:
This seems like a function that would be better suited for the SDK, so our API customers can use it as well. All of our other recovery flows exist in the SDK.

Reply:
I'm actually quite against having uncalled, untested recovery code in the SDK. This code should really live in it's own package, but we are not quite ready for that yet.

Reply:
We do have unit tests for the current recovery flows in BitGoJS. Not entirely sure what you mean by uncalled, the recovery functions live in either BaseCoin or AbstractUtxoCoin, so our API customers can call them by `bitgo.coin('btc').(recover|recoverFromWrongChain)()`

Reply:
Ah, my mistake. I was thinking of another piece of code which is uncalled and untested (downloadKeycard). It makes me feel better that the recovery code has at least happy-path unit tests, but I still think bitgojs is the wrong place for recovery code. The role of bitgojs is simply to bind the www REST API to js. Recovery lives at a higher level, which is using those bindings to do recoveries. That's why I think it should belong in a new package which should depend on bitgojs. Thoughts?

Reply:
I've been a fan of developing BitGoJS "modules" for a while, which would inject non-critical or niche methods into the greater package. I know that there are a few proposed modules (I can give more detail in private :P) but last I've heard we haven't come up with the logistics behind importing them. If we figure it out, I definitely think `bitgojs-recovery` would be an awesome module :D